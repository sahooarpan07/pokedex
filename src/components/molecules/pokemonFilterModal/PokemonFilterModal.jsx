import {View, Text, ScrollView, Pressable} from 'react-native';
import React, {useState} from 'react';
import PokemonModal from '../../atoms/modal/PokemonModal';
import AccordionItem from '../../atoms/accordionItem/AccordionItem';
import styles from './PokemonFilterModal.styles';
import PokemonChipsFilter from '../pokemonTypesFilter/PokemonTypesFilter';
const PokemonFilterModal = ({
  modalVisible,
  setModalVisible,
  updateFilterData,
}) => {
  const [pokemonTypes, setPokemonTypes] = useState([]);

  function handleUpdateFilterData() {
    const selectedTypes = pokemonTypes
      .filter(pokemon => pokemon.isChecked)
      .map(type => type.name);

    updateFilterData(selectedTypes);
    setModalVisible(!modalVisible);
  }

  return (
    <PokemonModal onRequestClose={setModalVisible} visible={modalVisible}>
      <View style={styles.outerContainer} testId="filterModal">
        <View style={styles.innerContainer}>
          {/* Header */}
          <View style={styles.header}>
            <Text style={styles.headerTitle}>Filters</Text>

            <Pressable style={styles.headerBtn} onPress={setModalVisible}>
              <Text style={styles.headerBtnText}>X</Text>
            </Pressable>
          </View>
          {/* AccordionÂ  */}
          <View
            style={styles.scrollContainer}
            showsVerticalScrollIndicator={false}>
            <AccordionItem title="Type">
              <PokemonChipsFilter
                pokemonTypes={pokemonTypes}
                setPokemonTypes={setPokemonTypes}
              />
            </AccordionItem>
          </View>
        </View>
        {/* ABSOLUTE BUTTON CONTAINER */}
        <View style={styles.btnContainer}>
          <Pressable
            testID=""
            style={[styles.btn, styles.btnSecondary]}
            onPress={() => {
              const updatedTypes = pokemonTypes.map(type => {
                return {
                  ...type,
                  isChecked: false,
                };
              });
              setPokemonTypes(updatedTypes);
            }}>
            <Text style={[styles.btnText, styles.btnSecondaryText]}>Reset</Text>
          </Pressable>
          <Pressable
            testID="apply"
            style={[styles.btn, styles.btnPrimary]}
            onPress={handleUpdateFilterData}>
            <Text style={[styles.btnText, styles.btnPrimaryText]}>Apply</Text>
          </Pressable>
        </View>
      </View>
    </PokemonModal>
  );
};

export default PokemonFilterModal;
