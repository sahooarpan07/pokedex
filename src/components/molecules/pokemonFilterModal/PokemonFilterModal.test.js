/**
 * @format
 */

import 'react-native';
import React from 'react';
import PokemonFilterModal from './PokemonFilterModal';
import {render, fireEvent, screen} from '@testing-library/react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders PokemonFilterModal correctly', () => {
  const tree = renderer
    .create(
      <PokemonFilterModal
        modalVisible={true}
        setModalVisible={jest.fn()}
        updateFilterData={jest.fn()}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('should test apply click', () => {
  render(
    <PokemonFilterModal
      modalVisible={true}
      setModalVisible={jest.fn()}
      updateFilterData={jest.fn()}
    />,
  );
  const applyButton = screen.getByTestId('apply');
  fireEvent.press(applyButton);
});
