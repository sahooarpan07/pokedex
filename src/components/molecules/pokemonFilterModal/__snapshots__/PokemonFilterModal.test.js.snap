// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokemonFilterModal correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "height": "100%",
      }
    }
    testId="filterModal"
  >
    <View
      style={
        {
          "padding": 24,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomColor": "#ddd",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 32,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#180345",
              "flex": 1,
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          Filters
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderColor": "#180345",
              "borderRadius": 20,
              "borderWidth": 1,
              "paddingHorizontal": 8,
              "paddingVertical": 2,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": 600,
              }
            }
          >
            X
          </Text>
        </View>
      </View>
      <View
        showsVerticalScrollIndicator={false}
        style={
          {
            "marginBottom": 150,
          }
        }
      >
        <View
          style={
            {
              "borderColor": "#180345",
              "borderRadius": 5,
              "borderWidth": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                {
                  "borderRightColor": "#ddd",
                  "borderRightWidth": 1,
                  "color": "#180345",
                  "flex": 1,
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              Type
            </Text>
            <Text
              style={
                {
                  "color": "#180345",
                  "flex": 2,
                  "fontSize": 18,
                  "paddingLeft": 10,
                }
              }
            >
              Status
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#180345",
                  "borderRadius": 30,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 8,
                  "paddingVertical": 2,
                }
              }
              testID="expand-button"
            >
              <Text
                style={
                  {
                    "color": "#180345",
                    "fontSize": 18,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#fff",
          "borderTopColor": "#ddd",
          "borderTopWidth": 1,
          "bottom": 0,
          "flexDirection": "row",
          "gap": 20,
          "paddingHorizontal": 20,
          "paddingVertical": 20,
          "position": "absolute",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            },
            {
              "borderColor": "#180345",
              "borderWidth": 1,
            },
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {},
              {
                "color": "#180345",
              },
            ]
          }
        >
          Reset
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            },
            {
              "backgroundColor": "#180345",
            },
          ]
        }
        testID="apply"
      >
        <Text
          style={
            [
              {},
              {
                "color": "#fff",
              },
            ]
          }
        >
          Apply
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
